src/processor/operator/shortest_path.cpp

```diff
@@ -0,0 +1,100 @@
+#include "processor/operator/shortest_path.h"
+
+#include "iostream"
+
+using namespace std;
+
+namespace kuzu {
+namespace processor {
+
+bool ShortestPathAdjList::getNextTuplesInternal() {
+    while (true) {
+        while (!bfsQueue.empty()) {
+            auto queueFrontNodeOffset = bfsQueue.front();
+            auto bfsFrontNodeState = bfsVisitedNodesMap[queueFrontNodeOffset];
```

Comment:
Is this a safe operation? 
1. You didn't initialize variables in BaseShortestPath, which is not a good habit. Because there can be unexpected initialized values on some platforms.
2. Why not first check if bfsVisitedNodesMap contains queueFrontNodeOffset, before you get the value (bfsFrontNodeState) from the map?

Reply:
1) I think the queue and map container always are initialized as empty, but anyway have added the initialization as part of the constructor now.

2) The map does not have any elements removed actually. And the only place it has elements added is the `addBfsLevel` function where the queue is also pushed with the element. So there is no case of the map missing any element. I've also added the map, queue proper initialization.